### PR TITLE
Add a guessing game with forced function calling

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

**Comment by natebosch** [comment]
cc @brianquinlan - I can't add you as a reviewer directly in this repo.

**Review by brianquinlan — COMMENTED** [review]
I think that you'll need approval from someone else.

File: samples/dart/bin/guessing_game.dart [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      tools: [
+        Tool(functionDeclarations: [
+          FunctionDeclaration(
+              'recordWords',
+              'Stores a list of words for a guessing game.',
+              Schema(SchemaType.object, properties: {
+                'words': Schema(SchemaType.array,
+                    items: Schema(SchemaType.string,
+                        description:
+                            'A Single word that a player will need to guess.'))
```
Comment (devoncarew) [inline samples/dart/bin/guessing_game.dart:40]:
```suggestion
                            'A single word that a player will need to guess.'))
```

File: samples/dart/bin/guessing_game.dart [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+              }))
+        ])
+      ],
+      toolConfig: ToolConfig(
+          functionCallingConfig:
+              FunctionCallingConfig(mode: FunctionCallingMode.any)));
```
Comment (devoncarew) [inline samples/dart/bin/guessing_game.dart:46]: A trailing comma here wouldn't hurt anything...

**Review by devoncarew — APPROVED** [review]
lgtm

(I assume the commits in `pkgs/google_generative_ai` have already landed and aren't really part of this review)

File: samples/dart/bin/guessing_game.dart (hunk 2 above)
Comment (natebosch) [reply samples/dart/bin/guessing_game.dart]: Done.

File: samples/dart/bin/guessing_game.dart (hunk 1 above)
Comment (natebosch) [reply samples/dart/bin/guessing_game.dart]: Done.

**Pull request merged**